packages/roosterjs-editor-core/lib/coreAPI/insertNode.ts

```diff
@@ -87,8 +99,13 @@ export const insertNode: InsertNode = (core: EditorCore, node: Node, option: Ins
 
             // Final check to see if the inserted node is a block. If not block and the ask is to insert on new line,
             // add a DIV wrapping
-            if (insertedNode && option.insertOnNewLine && !isBlockElement(insertedNode)) {
-                wrap(insertedNode);
+            if (insertedNode && option.insertOnNewLine) {
+                // When insert on new line, all the nodes to be inserted have been wrapped with a DIV in insertContent(),
+                // so just check the first node of insertedNode.
+                let wrappedNode = Array.isArray(insertedNode) ? insertedNode[0] : insertedNode;
+                if (!isBlockElement(wrappedNode)) {

```

Comment:
Need to check if the last node is block element as well. Otherwise, when the first one is block element but last one is not, it is possible the last node and the next one are in the same block

Reply:
The check for the last node is added.